include/nuttx/wireless/ieee80211/ieee80211.h

```diff
@@ -42,849 +42,4026 @@
+ * as BSS Channel Width
+ *
+ * @IEEE80211_S1G_CHANWIDTH_1MHZ: 1MHz operating channel
+ * @IEEE80211_S1G_CHANWIDTH_2MHZ: 2MHz operating channel
+ * @IEEE80211_S1G_CHANWIDTH_4MHZ: 4MHz operating channel
+ * @IEEE80211_S1G_CHANWIDTH_8MHZ: 8MHz operating channel
+ * @IEEE80211_S1G_CHANWIDTH_16MHZ: 16MHz operating channel
+ */
+
+enum ieee80211_s1g_chanwidth
+{
+  IEEE80211_S1G_CHANWIDTH_1MHZ  = 0,
+  IEEE80211_S1G_CHANWIDTH_2MHZ  = 1,
+  IEEE80211_S1G_CHANWIDTH_4MHZ  = 3,
+  IEEE80211_S1G_CHANWIDTH_8MHZ  = 7,
+  IEEE80211_S1G_CHANWIDTH_16MHZ = 15,
+};
+
+#define WLAN_SA_QUERY_TR_ID_LEN 2
+#define WLAN_MEMBERSHIP_LEN     8
+#define WLAN_USER_POSITION_LEN 16
+
+/* struct ieee80211_tpc_report_ie
+ *
+ * This structure refers to "TPC Report element"
+ */
+
+struct ieee80211_tpc_report_ie
+{
+  uint8_t tx_power;
+  uint8_t link_margin;
+};
+
+#define IEEE80211_ADDBA_EXT_FRAG_LEVEL_MASK   GENMASK(2, 1)
+#define IEEE80211_ADDBA_EXT_FRAG_LEVEL_SHIFT  1
+#define IEEE80211_ADDBA_EXT_NO_FRAG           BIT(0)
+
+struct ieee80211_addba_ext_ie
+{
+  uint8_t data;
+};
+
+/* struct ieee80211_s1g_bcn_compat_ie
+ *
+ * S1G Beacon Compatibility element
+ */
+
+struct ieee80211_s1g_bcn_compat_ie
+{
+  uint16_t compat_info;
+  uint16_t beacon_int;
+  uint32_t tsf_completion;
+};
+
+/* struct ieee80211_s1g_oper_ie
+ *
+ * S1G Operation element
+ */
+
+struct ieee80211_s1g_oper_ie
+{
+  uint8_t ch_width;
+  uint8_t oper_class;
+  uint8_t primary_ch;
+  uint8_t oper_ch;
+  uint16_t basic_mcs_nss;
+};
+
+/* struct ieee80211_aid_response_ie
+ *
+ * AID Response element
+ */
+
+struct ieee80211_aid_response_ie
+{
+  uint16_t aid;
+  uint8_t switch_count;
+  uint16_t response_int;
+};
+
+struct ieee80211_s1g_cap
+{
+  uint8_t capab_info[10];
+  uint8_t supp_mcs_nss[5];
+};
+
+struct ieee80211_ext
+{
+  uint16_t frame_control;
+  uint16_t duration;
+  union
+    {
+      struct
+        {
+          uint8_t sa[ETH_ALEN];
+          uint32_t timestamp;
+          uint8_t change_seq;
+          uint8_t variable[0];
+        } s1g_beacon;
+      struct
+        {
+          uint8_t sa[ETH_ALEN];
+          uint32_t timestamp;
+          uint8_t change_seq;
+          uint8_t next_tbtt[3];
+          uint8_t variable[0];
+        } s1g_short_beacon;
+    } u;
+};
+
+struct ieee80211_mgmt
+{
+  uint16_t frame_control;
+  uint16_t duration;
+  uint8_t da[ETH_ALEN];
+  uint8_t sa[ETH_ALEN];
+  uint8_t bssid[ETH_ALEN];
+  uint16_t seq_ctrl;
+  union
+    {
+      struct
+        {
+          uint16_t auth_alg;
+          uint16_t auth_transaction;
+          uint16_t status_code;
+
+          /* possibly followed by Challenge text */
+
+          uint8_t variable[0];
+        } auth;
+      struct
+        {
+          uint16_t reason_code;
+        } deauth;
+      struct
+        {
+          uint16_t capab_info;
+          uint16_t listen_interval;
+
+          /* followed by SSID and Supported rates */
+
+          uint8_t variable[0];
```

Comment:
I'm not sure if it is applicable here, but it is C89 incompatible

Reply:
@pkarashchenko since:

1. This file is controlled by 3rd party
2. It isn't a core header file included by many place
3. Chip which contain WiFi normally has the new version compiler

Can we relax the requirement here?